DFC.Digital.Tools/DFC.Digital.Tools.Service.Accounts/AccountsService.cs

```diff
@@ -0,0 +1,132 @@
+﻿using DFC.Digital.Tools.Core;
+using DFC.Digital.Tools.Data.Interfaces;
+using DFC.Digital.Tools.Data.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace DFC.Digital.Tools.Service.Accounts
+{
+    public class AccountsService : IAccountsService
+    {
+        private readonly IApplicationLogger applicationLogger;
+        private readonly IConfigConfigurationProvider configuration;
+        private readonly IAccountQueryRepository accountQueryRepository;
+        private readonly IAuditCommandRepository auditCommandRepository;
+        private readonly ICircuitBreakerQueryRepository circuitBreakerQueryRepository;
+        private readonly ICircuitBreakerCommandRepository circuitBreakerCommandRepository;
+
+        public AccountsService(
+                                IApplicationLogger applicationLogger,
+                                IConfigConfigurationProvider configuration,
+                                IAccountQueryRepository accountQueryRepository,
+                                IAuditCommandRepository auditCommandRepository,
+                                ICircuitBreakerQueryRepository circuitBreakerQueryRepository,
+                                ICircuitBreakerCommandRepository circuitBreakerCommandRepository)
+        {
+           this.applicationLogger = applicationLogger;
+           this.configuration = configuration;
+           this.accountQueryRepository = accountQueryRepository;
+           this.auditCommandRepository = auditCommandRepository;
+           this.circuitBreakerQueryRepository = circuitBreakerQueryRepository;
+           this.circuitBreakerCommandRepository = circuitBreakerCommandRepository;
+        }
+
+        public async Task<CircuitBreakerDetails> GetCircuitBreakerStatusAsync()
+        {
+            var circuitBreakerDetails = circuitBreakerQueryRepository.GetBreakerDetails();
+
+            if (circuitBreakerDetails == null)
+            {
+               applicationLogger.Trace("Adding default circuit breaker record on get as one does not exist");
+                return AddDefaultCircuitBreaker();
+            }
+            else if (circuitBreakerDetails.CircuitBreakerStatus == CircuitBreakerStatus.Open && circuitBreakerDetails.LastCircuitOpenDate.AddHours(24) < DateTime.Now)
```

Comment:
would have preferred this set in database i.e. retry after date, which would have helped us manipulate when to try and close the circuit